src/business-logic/services/configuration-service-implementation.ts

```diff
@@ -0,0 +1,18 @@
+import { Shelf } from '../../model/entities/shelf'
+import { ConfigurationService } from './interfaces/configuration-service'
+import { ConfigurationEventEmitter } from './interfaces/configuration-event-emitter'
+import { ShelfRepository } from '../../data-access/repositories/interfaces/shelf-repository'
+
+export class ConfigurationServiceImplementation implements ConfigurationService {
+
+  constructor(
+    private readonly configurationEventEmitter: ConfigurationEventEmitter,
+    private readonly shelfRepository: ShelfRepository
+  ) { }
+
+  public async updateShelf(shelf: Shelf): Promise<Shelf> {
+    const updateShelf: Shelf = await this.shelfRepository.updateShelf(shelf)
```

Comment:
I would argue this should be called `const updatedShelf` since it is the returned and updated shelf right?

Reply:
I agree. It's been updated.